robottelo/config/base.py

```diff
@@ -931,20 +931,42 @@ def __init__(self, *args, **kwargs):
         self.netmask = None
         self.gateway = None
         self.bridge = None
+        self.network = None
+        self.dhcp_ipam = None
+        self.dhcp_from = None
+        self.dhcp_to = None
+        self.dns_primary = None
 
     def read(self, reader):
         """Read Vlan Network settings."""
         self.subnet = reader.get('vlan_networking', 'subnet')
         self.netmask = reader.get('vlan_networking', 'netmask')
         self.gateway = reader.get('vlan_networking', 'gateway')
         self.bridge = reader.get('vlan_networking', 'bridge')
+        self.network = reader.get('vlan_networking', 'network')
+        self.dhcp_ipam = reader.get('vlan_networking', 'dhcp_ipam')
+        self.dhcp_from = reader.get('vlan_networking', 'dhcp_from')
+        self.dhcp_to = reader.get('vlan_networking', 'dhcp_to')
+        self.dns_primary = reader.get('vlan_networking', 'dns_primary')
 
     def validate(self):
         """Validate Vlan Network settings."""
         validation_errors = []
-        if not all(vars(self).values()):
+        if bool(self.bridge) == bool(self.network):
+            validation_errors.append(
+                'exactly one of the "bridge" or "network" parameters '
+                'must be specified')
+        if bool(self.dhcp_from) != bool(self.dhcp_to):
+            validation_errors.append(
+                'both or none of "dhcp_from", "dhcp_to" parameters '
+                'must be specified')
+            if self.dhcp_ipam and self.dhcp_ipam not in ['Internal DB', 'DHCP']:
+                validation_errors.append(
+                    '[vlan_networking] "dhcp_ipam" must be one of "Internal DB" or "DHCP"')
```

Comment:
Shouldn't this if be one level of indentation higher? Right now this will be checked only if exactly one of dhcp_from or dhcp_to is provided. But I think dhcp_ipam may be provided independently of dhcp_from/dhcp_to?

Reply:
you are right!
I added this one additionally as i originally forgot about it. And i did not check the indent level properly in a hassle.

Reply:
@rplevka no problem! That's why we do these code reviews.